corehq/util/soft_assert/core.py

```diff
@@ -0,0 +1,57 @@
+from collections import namedtuple
+import hashlib
+import traceback
+
+
+SoftAssertInfo = namedtuple('SoftAssertInfo',
+                            ['traceback', 'count', 'msg', 'key', 'line',
+                             'short_traceback'])
+
+
+class SoftAssert(object):
+    def __init__(self, debug=False, send=None, incrementing_counter=None,
+                 should_send=None, skip_frames=0, key_limit=2):
+        assert send
+        assert incrementing_counter
+        assert should_send
+        self.debug = debug
+        self.send = send
+        self.incrementing_counter = incrementing_counter
+        self.should_send = should_send
+        self.tb_skip = skip_frames + 3
+        self.key_limit = key_limit
+
+    def __call__(self, assertion, msg=None):
+        return self._call(assertion, msg)
+
+    call = __call__
```

Comment:
gotcha, my gut says the first one since more often than not the soft asserts for a given soft_assert scope will be the same recipients and _assert() seems more pythonic. cool though, definitely don't mind leaving them both in